src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -92,6 +93,13 @@ private static boolean isValidPath(Path filePath) {
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
+        
+        // XXX handling of file deleted situation using exceptions.
+        // I think this is a very bad idea. Instead of handling file deleted situation, the
+        // following code will just check if the file exists before saving
+        if (!Files.exists(path)) {
+            throw new StorageOperationException("File does not exist!");
+        }
```

Comment:
checking if the file exists before saving is still make sense because the file might corrupt while the user is editing the program.  Like what you have pointed out, this would degrade the user experience, could you think of other better way in solving it?

Note that, we can also use exception as a warning, instead of stopping the operation ^^